### PR TITLE
[FSTORE-850] DBT tutorial with BigQuery

### PR DESCRIPTION
**Pull request by Maxxx-zh**

### CODE REVIEW

**Comment by jimdowling** [comment]
Can you rename the directory from 'dbt' to 'dbt_bq' ?

**Comment by jimdowling** [comment]
Is it possible to set the hopsworks ENV variables in the dbt_project.yml file?
 HOPSWORKS_PROJECT, HOPSWORKS_HOST
 What about HOPSWORKS_API_KEY - where should that be set?

**Comment by Maxxx-zh** [comment]
> Did you try inserting a PySpark dataframe into Hopsworks, instead of a Pandas DF - which is much slower?

Got the next error:
`typeerror: The provided dataframe type is not recognized. Supported types are: pandas dataframe. The provided dataframe has type: <class 'pyspark.sql.dataframe.DataFrame'>`

**Comment by Maxxx-zh** [comment]
> Is it possible to set the hopsworks ENV variables in the dbt_project.yml file?
> HOPSWORKS_PROJECT, HOPSWORKS_HOST
> What about HOPSWORKS_API_KEY - where should that be set?

Знімок екрана 2023-05-10 о 23 34 42

File: integrations/dbt_bq/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+## <span style='color:#ff5f27'> 👨🏻‍🏫 DBT Tutorial with BigQuery </span>
+
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:2]: This tutorial shows you how to perform feature engineering in DBT on BigQuery, storing offline computed features in a table in BigQuery (that is mounted as an external feature group in Hopsworks) and online features in Hopsworks. The online features are written to Hopsworks using a Python module that is run on a DataProc cluster. The feature group created in Hopsworks has its offline data stored in BigQuery and its online data stored in Hopsworks online store (RonDB).

File: integrations/dbt_bq/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+### <span style='color:#ff5f27'> 🏡 Cluster setup </span>
+
+To begin with, it's necessary to setup the Dataproc cluster.
+
+You need to configure an external Spark cluster to be able to interact with the Hopsworks Feature Store.
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:7]: First, you need to setup a Dataproc (Spark) cluster that will run the Python model in our DBT workflow. The Python model will write to the online feature store in Hopsworks.

File: integrations/dbt_bq/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+Navigate to **Project Settings** and then **Integrations**. At the bottom of the page you will find necessary files which you need to attach to your Dataproc cluster.
+
+![output](images/sparkConfig.png)
+
+Code to create Dataproc cluster is present in `provision.sh` file.
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:13]: You can find the code to create the Dataproc cluster in `provision.sh`.

File: integrations/dbt_bq/README.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+To make `provision.sh` file executable do the next:
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:15]: executable, run the following command:

File: integrations/dbt_bq/README.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+`chmod +x provision.sh`
+
+Fill in your cluster information and then use `./provision.sh` command to start the cluster creation.
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:19]: then run the `./provision.sh` command

File: integrations/dbt_bq/README.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+### <span style='color:#ff5f27'>📡 DBT Setup </span>
+
+Install the BigQuery adapter running
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:23]: by running

File: integrations/dbt_bq/README.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+`pip install dbt-bigquery`
+
+Create a new profile inside your ~/.dbt/profiles.yml file.
+
+```
+{YOUR_PROJECT_NAME}:
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:29]: YOUR_GCP_PROJECT_NAME
or 
YOUR_HOPSWORKS_PROJECT_NAME
or
YOUR_DBT_PROJECT_NAME

File: integrations/dbt_bq/README.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+ target: dev
+ outputs:
+   dev:
+     # Type of DBT connector (BigQuery, Snowflake, etc)
+     type: bigquery
+     # Authentication method 
+     method: service-account-json
+     # Your Google Cloud project name
+     project: [your-project-id]
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:38]: your-gcp-project-id

File: integrations/dbt_bq/README.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+```
+{YOUR_PROJECT_NAME}:
+ target: dev
+ outputs:
+   dev:
+     # Type of DBT connector (BigQuery, Snowflake, etc)
+     type: bigquery
+     # Authentication method 
+     method: service-account-json
+     # Your Google Cloud project name
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:37]: it's the project id, not the project name, right?

File: integrations/dbt_bq/README.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+     # Your Bucket name
+     gcs_bucket: {YOUR_BUCKET_NAME}
+     # Your Dataproc region
+     dataproc_region: {YOUR_DATAPROC_REGION} 
+ ```
+
+
+### <span style='color:#ff5f27'>⚙️ DBT Launch </span>
+
+Fill in `read_bigquery_data.sql` and `data_pipeline.py` files with your information.
```
Comment (jimdowling) [inline integrations/dbt_bq/README.md:67]: with your feature engineering code that creates features and writes them to the BQ offline table and Hopsworks online table.

**Review by jimdowling — APPROVED** [review]
Fixes in the readme

**Review by jimdowling — APPROVED** [review]
Fixes in the readme

File: integrations/dbt_bq/provision.sh [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    --bucket $BUCKET_NAME \
+    --region $REGION \
+    --zone europe-central2-c \
+    --master-machine-type n1-standard-2 \
+    --master-boot-disk-size 50GB \
+    --num-workers 2 \
+    --worker-machine-type n1-standard-2 \
+    --worker-boot-disk-size 50GB \
+    --image-version 2.0-debian10 \
+    --optional-components JUPYTER \
```
Comment (davitbzh) [inline integrations/dbt_bq/provision.sh:18]: This is not needed

**Review by davitbzh — CHANGES_REQUESTED** [review]

**Review by davitbzh — APPROVED** [review]

**Pull request merged**